### PR TITLE
enable travis-ci

### PR DESCRIPTION
**Pull request by kr**

### CODE REVIEW

**Comment by kr** [comment]
PTAL

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+language: go
+
+go: 1.4
```
Comment (freeformz) [inline .travis.yml]: Not 1.4.2 ? Or does 1.4 imply the latest from the 1.4.X line?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+script:
+    - git config --global user.email "you@example.com"
+    - git config --global user.name "Your Name"
```
Comment (freeformz) [inline .travis.yml]: Why is this?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    - test -z "$(go fmt)"
+    - go vet
+    - go test
+    - go test -race
```
Comment (freeformz) [inline .travis.yml]: Have you considered golint as well?

**Comment by freeformz** [comment]
LGTM modulo questions.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+language: go
+
+go: 1.4
+
+script:
+    - git config --global user.email "you@example.com"
+    - git config --global user.name "Your Name"
+    - test -z "$(go fmt)"
+    - go vet
+    - go test
```
Comment (freeformz) [inline .travis.yml]: I usually add `-v` to my .travis.yml files as well.

**Comment by freeformz** [comment]
@kr I'm going to merge this so things stop failing as Travis seems to default to ruby.

File: .travis.yml (hunk 2 above)
Comment (kr) [inline .travis.yml]: Godep's unit tests run git to synthesize repositories on disk, and git (at least on travis) requires these to be configured.

File: .travis.yml (hunk 3 above)
Comment (kr) [inline .travis.yml]: Let's add golint in a separate commit, along with fixing the dozen or so things it complains about.

**Comment by kr** [comment]
Uhh, it would have been nice to be able to address your feedback before we merged this. If we don't want travis complaining while this PR is still open the fix is to turn on this option:

build only if travis.yml is present

I just turned it on.

**Comment by kr** [comment]
Sorry, I guess I had 2 days' worth of chance to address the feedback. :sweat_smile:

My bad.

**Comment by freeformz** [comment]
@kr Also my bad. Was kind of trigger happy.

**Pull request merged**